api/src/main/java/io/druid/segment/loading/DataSegmentArchiver.java

```diff
@@ -21,8 +21,31 @@
 
 import io.druid.timeline.DataSegment;
 
+import javax.annotation.Nullable;
+
 public interface DataSegmentArchiver
 {
-  public DataSegment archive(DataSegment segment) throws SegmentLoadingException;
-  public DataSegment restore(DataSegment segment) throws SegmentLoadingException;
+  /**
+   * Perform an archive task on the segment and return the resulting segment or null if there was no action needed.
+   *
+   * @param segment The source segment
+   *
+   * @return The segment after archiving or `null` if there was no archiving performed.
+   *
+   * @throws SegmentLoadingException on error
+   */
+  @Nullable
+  DataSegment archive(DataSegment segment) throws SegmentLoadingException;
```

Comment:
That's pretty challenging since `loadSpec` is just a `Map<String, Object>` which may have whatever junk in it. I opted for this way of doing it so that the implementation can make the determination on if the segment was moved or not.